core/core-impl/src/main/java/ai/timefold/solver/core/impl/domain/variable/listener/support/VariableListenerSupport.java

```diff
@@ -94,7 +102,9 @@ public <Supply_ extends Supply> boolean cancel(Demand<Supply_> demand) {
         var supplyWithDemandCount = supplyMap.get(demand);
         if (supplyWithDemandCount == null) {
             return false;
-        } else if (supplyWithDemandCount.demandCount == 1L) {
+        }
+        demand.cancel(this);
```

Comment:
Since it increases the demand count, I'd say it has to be.
Practically it doesn't matter, because demands are rarely cancelled - the ones where it matters (the transitive demands) are _never_ cancelled.